tape/src/main/java/com/squareup/tape/QueueFile.java

```diff
@@ -462,6 +459,31 @@ public synchronized void forEach(ElementVisitor reader) throws IOException {
     }
   }
 
+  /**
+   * First reads header to update element counts and file length. Then Scans each element in the
+   * queue for its length. To be used for Integrity Scans to detect queue corruption.
+   *
+   * @return int[] of all segment lengths.
+   * @throws IOException if there is a problem scanning the file (assume corruption).
+   */
+  public synchronized int[] scanAll() throws IOException {
+    // Read header to update all our counts
+    readHeader();
+    int position = first.position;
+    final int[] lengths = new int[elementCount];
+    int totalSize = HEADER_LENGTH;
+    for (int i = 0; i < elementCount; i++) {
+      Element current = readElement(position);
+      lengths[i] = current.length;
+      totalSize += current.length + Element.HEADER_LENGTH;
+      position = wrapPosition(current.position + Element.HEADER_LENGTH + current.length);
+    }
+    if (totalSize > fileLength || totalSize > raf.length() ) {
+      throw new IOException("Element scan reports size larger than file.");
+    }
+    return lengths;
```

Comment:
Returning lengths isn't super useful here, since the case we're looking for should lead to either an IOException while reading or an incorrect size. Basically this will only return when everything is fine and we don't really care.